### ISSUE TITLE
Architectural issues

### ISSUE THREAD

**Issue by NotAFile** [piqueserver/piqueserver#90]
The original author had a very strong fondness for subclassing. However this is beyond pointless, since those classes are huge singletons and inhereted from only once. This makes the code very hard to follow since the code might be in two or more locations. I presume the intent was seperation of concerns: the author wanted to seperate the AoS protocol implementation from the server implementation. However the seperation is not strong enough to make this actually useful. The following should be done:

- [ ]  decrease class size
- [ ]  merge classes and subclasses where useful (prime example: `FeatureTeam`)
- [ ]  remove unused functions

It might be hard to do this at times because of the plugin model of pyspades. By removing unused functions we might break plugins we don't know of.

**Comment by noway** [comment]
Yeah, breaking compatibility is not very fun, otherwise good idea.

**Comment by NotAFile** [comment]
Closing this since it's not specific enough
